modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -2726,78 +2733,66 @@ private synchronized void stop0(boolean cancel, boolean waitForBackups) {
 
                         GridDhtPartitionFullMap fullMap = grpCtx.topology().partitionMap(false);
 
-                        int cacheSpecificAmountOfOwners = -1;
+                        if (fullMap == null) {
+                            safeToStop = false;
 
-                        if (fullMap != null) {
-                            nodesToExclude.retainAll(fullMap.keySet());
+                            break;
+                        }
 
-                            int parts = grpCtx.topology().partitions();
+                        nodesToExclude.retainAll(fullMap.keySet());
 
-                            for (int part = 0; part < parts; part++) {
-                                Set<UUID> partIdealAssignment = grpCtx.affinity().idealAssignmentRaw().get(part).
-                                    stream().map(ClusterNode::id).collect(Collectors.toSet());
+                        if (!haveCopyLoclaPartitions(grpCtx, nodesToExclude, proposedSuppliers)) {
+                            safeToStop = false;
 
-                                int cnt = 0;
+                            break;
+                        }
 
-                                for (Map.Entry<UUID, GridDhtPartitionMap> entry : fullMap.entrySet()) {
-                                    if (nodesToExclude.contains(entry.getKey()))
-                                        continue;
+                        if (!isRebalanceCompleted(grpCtx)) {
+                            safeToStop = false;
```

Comment:
Did you consider adding log message here? Probably with throttling, at least in case of small WAIT_FOR_BACKUPS_CHECK_INTERVAL.

Reply:
Added log message here and another one in case of no other owners of local node partitions.